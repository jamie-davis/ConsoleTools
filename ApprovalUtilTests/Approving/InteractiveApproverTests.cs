using ApprovalUtil.Approving;
using ApprovalUtil.Scanning;
using ApprovalUtilTests.TestDoubles;
using ConsoleToolkit.ConsoleIO;
using ConsoleToolkit.Testing;
using TestConsoleLib.Testing;

namespace ApprovalUtilTests.Approving;

public class InteractiveApproverTests
{
    [Fact]
    public void InteractiveApproverDescribesResultsAndShowsMenu()
    {
        //Arrange
        var testConsole = new UnitTestConsole(typeof(Program).Namespace);
        var console = testConsole.Console;
        var error = testConsole.Error;

        ApprovalTestResult MakeTestResult(string testTypeName, string testName, bool passed)
        {
            var approvedFile = $"{testTypeName}.{testName}.approved.txt";
            var receivedFile = $"{testTypeName}.{testName}.received.txt";
            var sourceFile = $"{testTypeName}.cs";
            var approvalTestOutput = new ApprovalTestOutput(sourceFile, approvedFile, receivedFile, testTypeName, testName);
            return new ApprovalTestResult(approvalTestOutput, passed);
        }

        var tests = new List<ApprovalTestResult>()
        {
            MakeTestResult("TestClass", "TestOne", true),
            MakeTestResult("TestClass", "TestTwo", true),
            MakeTestResult("TestClass", "TestThree", false),
            MakeTestResult("TestClass", "TestFour", true),
        };
        
        var approverParams = new FakeApproverParams { Interactive = true };
        var data = @"X
";
        using var stream = new StringReader(data);
        testConsole.Interface.SetInputStream(stream);

        //Act
        var _ = InteractiveApprover.Execute(console, error, tests, approverParams);

        //Assert
        testConsole.Interface.GetBuffer(ConsoleBufferFormat.TextOnly).Verify();
    }
}