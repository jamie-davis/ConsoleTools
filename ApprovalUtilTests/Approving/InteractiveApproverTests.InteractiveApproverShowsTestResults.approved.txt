Located 7 tests. 3 failures, 4 passes                       
                                                            
Interactive functions                                       
                                                            
1-View TestClass1.TestFour failure                          
2-View TestClass2.TestTwo failure                           
3-View TestClass2.NewTest failure                           
A-Approve all and exit                                      
X-Exit                                                      
                                                            
Select an option: 1                                         
                                                            
=================                                           
>> ---                                                      
This·is·a·test·approved·file¶                               
It·contains·compare·data·for·tests·to·fi                    
nd.¶                                                        
                                                            
~~~~~~~~~~~~~~~~~~~~                                        
>> +++                                                      
This·is·a·test·received·file¶                               
It·contains·compare·data·for·tests·to·fi                    
nd.¶                                                        
=================                                           
Approval functions                                          
                                                            
A-Approve                                                   
X-Done                                                      
                                                            
Select an option: X                                         
                                                            
Interactive functions                                       
                                                            
1-View TestClass1.TestFour failure                          
2-View TestClass2.TestTwo failure                           
3-View TestClass2.NewTest failure                           
A-Approve all and exit                                      
X-Exit                                                      
                                                            
Select an option: X                                         
                                                            